doc: dev model: Add a reference to ncs-example-application (#9770)
nrfconnect/sdk-nrf: doc/nrf/dm_adding_code.rst

In order to implement this approach, you first need to create a manifest repository of your own, which just means a repository that contains a :file:`west.yml` manifest file in its root.
Next you must populate the manifest file with the list of repositories and their revisions.

> In order to create your own repository you have a couple of options, described in the sections below.

Reviewer: ```suggestion
You have a couple of different options to create your own repository.
```

Once you have your new manifest repository hosted online, you can use it with west just like you use the `sdk-nrf`_  repository when :ref:`getting <dm-wf-get-ncs>` and later :ref:`updating <dm-wf-update-ncs>` the source code.
You just need to replace ``sdk-nrf`` and ``nrf`` with the repository name and path you have chosen for your manifest repository (*your-name/your-application* and *your-app-workspace*, respectively), as shown in the following code:

.. parsed-literal::
   :class: highlight

   west init -m https:\ //github.com/*your-name/your-application* *your-app-worskpace*
   cd *your-app-workspace*
   west update

After that, to modify the |NCS| version associated with your app, change the ``revision`` value in the manifest file to the `sdk-nrf`_ Git tag, SHA, or the branch you want to use, save the file, and run ``west update``.
See :ref:`zephyr:west-basics` for more details.

> Recommended: Using the example application repo
> -----------------------------------------------

Reviewer: ```suggestion
Recommended: Using the example application repository
-----------------------------------------------------
```

> Nordic maintains a templated example repository for the purpose of serving as a reference for users that choose this workflow, `ncs-example-application`_.

Reviewer: ```suggestion
Nordic Semiconductor maintains a templated example repository as a reference for users that choose this workflow, `ncs-example-application`_.
```

> This repository showcases many features of both the |NCS| and Zephyr:

Reviewer: ```suggestion
This repository showcases the following features of both the |NCS| and Zephyr:
```

* Basic :ref:`Zephyr application <zephyr:application>` skeleton
* :ref:`Zephyr workspace applications <zephyr:zephyr-workspace-app>`
* :ref:`West T2 topology <zephyr:west-t2>`
* :ref:`Custom boards <zephyr:board_porting_guide>`
* Custom :ref:`devicetree bindings <zephyr:dt-bindings>`
* Out-of-tree :ref:`drivers <zephyr:device_model_api>`
* Out-of-tree libraries
* Example CI configuration (using Github Actions)
* Custom :ref:`west extension <zephyr:west-extensions>`

> It is highly recommended to use this templated repository as a starting point for
> your own application manifest repo. You can do so by clicking the "Use this
> template" button on the GitHub web user interface.
> Once you have your own copy of the repository you can then modify it at will,
> including adding your own projects to the manifest file, as well as your own boards,
> drivers, and libraries as required by your application.

Reviewer: ```suggestion
It is highly recommended to use this templated repository as a starting point for your own application manifest repository.
Click the :guilabel:`Use this template` button on the GitHub web user interface.
Once you have your own copy of the repository, you can then modify it.
You can add your own projects to the manifest file, as well as your own boards, drivers, and libraries as required by your application.
```

> The `ncs-example-application`_ repository is tagged every time a new release of
> the |NCS| is made (starting with NCS v2.3.0), using the exact same version
> number. This allows an application developer to select the tag that will match
> the version of the |NCS| that they intend to use.

Reviewer: ```suggestion
The `ncs-example-application`_ repository is tagged every time a new release of the |NCS| is launched (starting with v2.3.0), using the same version number.
This allows you to select the tag that matches the version of the |NCS| you intend to use.
```

> Creating your own manifest repo from scratch
> --------------------------------------------

Reviewer: ```suggestion
Creating your own manifest repository from scratch
--------------------------------------------------
```

> If, for whatever reason you decide not to use the example application repository
> as a starting point, then you can start from scratch to create your own manifest repo that imports the |NCS|.

Reviewer: ```suggestion
If you decide not to use the example application repository as a starting point, you can start from scratch to create your own manifest repository that imports the |NCS|.
```

> In general, the easiest thing to do is to ``import`` `sdk-nrf`_ in your own :file:`west.yml`, using west's :ref:`manifest imports <zephyr:west-manifest-import>` feature.
> This is demonstrated by the following code, that would be placed somewhere in
> your repository (typically in :file:`root/west.yml`):

Reviewer: ```suggestion
To start the process, ``import`` `sdk-nrf`_ in your own :file:`west.yml`, using the :ref:`manifest imports <zephyr:west-manifest-import>` feature of west.
This is demonstrated by the following code, that would be placed somewhere in your repository (typically in :file:`root/west.yml`):
```
